packages/cli/src/runDeploy.ts

```diff
@@ -0,0 +1,122 @@
+  const profile = opts.profile ?? process.env.FOUNDRY_PROFILE;
+
+  const config = (await loadConfig(opts.configPath)) as StoreConfig & WorldConfig;
+  if (opts.printConfig) {
+    console.log(chalk.green("\nResolved config:\n"), JSON.stringify(config, null, 2));
+  }
+
+  const srcDir = opts.srcDir ?? (await getSrcDirectory(profile));
+  const outDir = await getOutDirectory(profile);
+  const remappings = await getRemappings();
+
+  const rpc = opts.rpc ?? (await getRpcUrl(profile));
+  console.log(
+    chalk.bgBlue(
+      chalk.whiteBright(`\n Deploying MUD contracts${profile ? " with profile " + profile : ""} to RPC ${rpc} \n`)
+    )
+  );
+
+  if (opts.clean) {
+    await forge(["clean"], { profile });
+  }
+
+  // Run forge build
+  if (!opts.skipBuild) {
+    const outPath = path.join(srcDir, config.codegenDirectory);
+    await tablegen(config, outPath, remappings);
+    await forge(["build", "--skip", "test", "script"], { profile });
+    await execa("mud", ["abi-ts"], { stdio: "inherit" });
+  }
+
+  const privateKey = process.env.PRIVATE_KEY as Hex;
+  if (!privateKey) {
+    throw new MUDError(
+      `Missing PRIVATE_KEY environment variable.
+Run 'echo "PRIVATE_KEY=0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80" > .env'
+in your contracts directory to use the default anvil private key.`
+    );
+  }
+
+  const resolvedConfig = resolveConfig({ config, forgeSourceDir: srcDir, forgeOutDir: outDir });
+
+  const client = createWalletClient({
+    transport: http(rpc),
+    account: privateKeyToAccount(privateKey),
+  });
+  console.log("Deploying from", client.account.address);
+
+  const startTime = Date.now();
+  const worldDeploy = await deploy({
+    worldAddress: opts.worldAddress as Hex | undefined,
+    client,
+    config: resolvedConfig,
+  });
+  await postDeploy(config.postDeployScript, worldDeploy.address, rpc, profile);
+  console.log(chalk.green("Deployment completed in", (Date.now() - startTime) / 1000, "seconds"));
+
+  if (opts.saveDeployment) {
+    const deploymentInfo = {
+      worldAddress: worldDeploy.address,
+      blockNumber: Number(worldDeploy.fromBlock),
+    };
+    const chainId = await getChainId(client);
+    const deploysDir = path.join(config.deploysDirectory, chainId.toString());
+    mkdirSync(deploysDir, { recursive: true });
+    writeFileSync(path.join(deploysDir, "latest.json"), JSON.stringify(deploymentInfo, null, 2));
+    writeFileSync(path.join(deploysDir, Date.now() + ".json"), JSON.stringify(deploymentInfo, null, 2));
+
+    const localChains = [1337, 31337];
+    const deploys = existsSync(config.worldsFile) ? JSON.parse(readFileSync(config.worldsFile, "utf-8")) : {};
+    deploys[chainId] = {
+      address: deploymentInfo.worldAddress,
+      // We expect the worlds file to be committed and since local deployments are often a consistent address but different block number, we'll ignore the block number.
```

Comment:
nit (trying to avoid horizontal scroll bars and better vim support)
```suggestion
      // We expect the worlds file to be committed and since local deployments are often
      // a consistent address but different block number, we'll ignore the block number.
```